src/Microsoft.AspNetCore.Mvc.ViewFeatures/Controller.cs

```diff
@@ -102,7 +102,7 @@ public dynamic ViewBag
         [NonAction]
         public virtual ViewResult View()
         {
-            return View(viewName: null);
+            return View(viewName: ControllerContext.ActionDescriptor.ActionName);
```

Comment:
@javiercn the current approach doesn't work end-to-end.

Your suggested approach _might_ work (if the action name is always correct in the route values, even when using attribute routing). But does the value of removing `ActionDescriptor.Name` justify that large a change?